### PR TITLE
iina: add wrapper for iina-cli

### PR DESCRIPTION
**Pull request by YvesStraten**
## Description of changes
This PR adds a wrapper for the iina-cli binary to `$out/bin`, mimicking the brew cask. This binary is needed by applications such as [ani-cli](https://github.com/pystardust/ani-cli) as they rely on it to be present in PATH.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/video/iina/default.nix [hunk 1]
```diff
@@ -3,8 +3,8 @@
 , stdenv
 , undmg
 , nix-update-script
+,
```
Comment (stepbrobd) [inline pkgs/applications/video/iina/default.nix:6]:
```suggestion

```

File: pkgs/applications/video/iina/default.nix [hunk 2]
```diff
@@ -3,8 +3,8 @@
...
 }:
```
Comment (stepbrobd) [inline pkgs/applications/video/iina/default.nix:7]:
```suggestion
}:

```

File: pkgs/applications/video/iina/default.nix [hunk 3]
```diff
@@ -19,8 +19,10 @@ stdenv.mkDerivation rec {
   sourceRoot = "IINA.app";
 
   installPhase = ''
+    mkdir -p "$out/bin"
     mkdir -p "$out/Applications/IINA.app"
```
Comment (stepbrobd) [inline pkgs/applications/video/iina/default.nix:23]:
```suggestion
```

File: pkgs/applications/video/iina/default.nix [hunk 4]
```diff
@@ -19,8 +19,10 @@ stdenv.mkDerivation rec {
   sourceRoot = "IINA.app";
 
   installPhase = ''
+    mkdir -p "$out/bin"
```
Comment (stepbrobd) [inline pkgs/applications/video/iina/default.nix:22]:
```suggestion
    mkdir -p $out/{bin,Applications/IINA.app}
```

**Review by stepbrobd — CHANGES_REQUESTED** [review]
seems like [IINA 1.3.4](https://github.com/iina/iina/releases/tag/v1.3.4) was out 2 days ago

**Comment by stepbrobd** [comment]
everything else looks good

Result of `nixpkgs-review pr 278174` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>iina</li>
  </ul>
</details>

**Comment by YvesStraten** [comment]
> seems like [IINA 1.3.4](https://github.com/iina/iina/releases/tag/v1.3.4) was out 2 days ago

I did not update the version because usually nixpkgs wants one pr for one purpose

**Comment by YvesStraten** [comment]
I will add your requested changes as well, please wait a moment

**Comment by stepbrobd** [comment]
Squash those two commits and we are good to go, happy new year btw 🎉

**Comment by YvesStraten** [comment]
Will do, thank you so much, happy new year to you too!

**Comment by stepbrobd** [comment]
Result of `nixpkgs-review pr 278174` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>iina</li>
  </ul>
</details>

**Review by stepbrobd — APPROVED** [review]

**Comment by Luflosi** [comment]
I updated the package in #280297.

**Pull request merged**